src/Dfe.Spi.RatesAdapter.Application/RatesManager.cs

```diff
@@ -1,21 +1,147 @@
 ﻿namespace Dfe.Spi.RatesAdapter.Application
 {
+    using System;
     using System.Threading;
     using System.Threading.Tasks;
     using Dfe.Spi.Models.Entities;
+    using Dfe.Spi.Models.RatesModels;
     using Dfe.Spi.RatesAdapter.Application.Definitions;
+    using Dfe.Spi.RatesAdapter.Domain.Definitions;
+    using LocalDomainModels = Dfe.Spi.RatesAdapter.Domain.Models;
 
     /// <summary>
     /// Implements <see cref="IRatesManager" />.
     /// </summary>
     public class RatesManager : IRatesManager
     {
+        private readonly ISchoolInformationStorageAdapter schoolInformationStorageAdapter;
+
+        /// <summary>
+        /// Initialises a new instance of the <see cref="RatesManager" />
+        /// class.
+        /// </summary>
+        /// <param name="schoolInformationStorageAdapter">
+        /// An instance of type
+        /// <see cref="ISchoolInformationStorageAdapter" />.
+        /// </param>
+        public RatesManager(
+            ISchoolInformationStorageAdapter schoolInformationStorageAdapter)
+        {
+            this.schoolInformationStorageAdapter = schoolInformationStorageAdapter;
+        }
+
         /// <inheritdoc />
-        public Task<Rates> GetRatesAsync(
-            string id,
+        public async Task<Rates> GetRatesAsync(
+            int year,
+            string entityName,
+            string identifier,
             CancellationToken cancellationToken)
         {
-            throw new System.NotImplementedException();
+            Rates toReturn = null;
+
+            if (entityName == nameof(LearningProvider))
+            {
+                long urn;
+                if (!long.TryParse(identifier, out urn))
+                {
+                    // TODO: Throw some sort of exception.
+                }
+
+                LocalDomainModels.SchoolInformation schoolInformation =
+                    await this.schoolInformationStorageAdapter.GetSchoolInformationAsync(
+                        year,
+                        urn,
+                        cancellationToken)
+                        .ConfigureAwait(false);
+
+                toReturn = Map(schoolInformation);
+            }
+            else
+            {
+                throw new NotImplementedException(
+                    "Not yet done! Implement me!");
```

Comment:
While in here, probably make this something more obvious, like "Rates for anything but LearningProvider is not supported yet". Just so we know what it is if we get it in the logs

Reply:
Sounds good, going to polish it up today 👍